rct229/rulesets/ashrae9012019/section1/section1rule3.py

```diff
@@ -0,0 +1,154 @@
+from rct229.rule_engine.ruleset_model_factory import produce_ruleset_model_instance
+from rct229.rulesets.ashrae9012019 import (
+    BASELINE_0,
+    BASELINE_90,
+    BASELINE_180,
+    BASELINE_270,
+    PROPOSED,
+    USER,
+)
+from rct229.utils.assertions import assert_
+from rct229.utils.jsonpath_utils import find_one
+from rct229.utils.std_comparisons import std_equal
+
+
+class Section1Rule3(RuleDefinitionBase):
+    """Rule 3 of ASHRAE 90.1-2019 Appendix G Section 1 (Performance Calculations)"""
+
+    def __init__(self):
+        super(Section1Rule3, self).__init__(
+            rmrs_used=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
+            rmrs_used_optional=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
+            id="1-3",
+            description="The Performance Cost Index-Target (PCIt) shall be calculated using the procedures defined in Section 4.2.1.1. "
+            "The PCIt shall be equal to [baseline building unregulated energy cost (BBUEC) + BPF x baseline building regulated energy cost (BBREC)]/ BBP",
+            ruleset_section_title="Performance Calculations",
+            standard_section="Section 4.2.1.1",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+        )
+
+    def get_calc_vals(self, context, data=None):
+        rmd_u = context.USER
+        rmd_b0 = context.BASELINE_0
+        rmd_b90 = context.BASELINE_90
+        rmd_b180 = context.BASELINE_180
+        rmd_b270 = context.BASELINE_270
+        rmd_p = context.PROPOSED
+
+        pci_target_set = []
+        bpf_set = []
+        bbp_set = []
+        bbrec_set = []
+        bbuec_set = []
+        for rmd in (rmd_u, rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p):
+            if rmd is not None:
+                pci_target_set.append(
+                    find_one("$.output.performance_cost_index_target", rmd)
+                )
+                bpf_set.append(
+                    find_one(
+                        "$.output.total_area_weighted_building_performance_factor",
+                        rmd,
+                    )
+                )
+                bbp_set.append(
+                    find_one("$.output.baseline_building_performance_energy_cost", rmd)
+                )
+                bbrec_set.append(
+                    find_one("$.output.baseline_building_regulated_energy_cost", rmd)
+                )
+                bbuec_set.append(
+                    find_one("$.output.baseline_building_unregulated_energy_cost", rmd)
+                )
```

Comment:
After the `for` loop, you may need to add an assertion to make sure those lists are not empty.

```python
assert_(pci_target_set and bpf_set and bbp_set and bbrec_set and bbuec_set, "Required data is missing in the RDS, make sure all RDSs have the following data: output.performance_cost_index_target, ....")
```

Reply:
The assert_ messages are available in lines 86-93.